Shared/sdk/CScriptArgReader.h

```diff
@@ -1189,10 +1190,10 @@ class CScriptArgReader
         return (NextCouldBeNumber() && NextCouldBeNumber(1) && NextCouldBeNumber(2) && NextCouldBeNumber(3)) || NextIsUserDataOfType<CLuaVector4D>();
     }
 
-    bool NextIsVector3D() const
+    bool NextIsVector3D(int iOffset = 0) const
     {
-        return (NextCouldBeNumber() && NextCouldBeNumber(1) && NextCouldBeNumber(2)) || NextIsUserDataOfType<CLuaVector3D>() ||
-               NextIsUserDataOfType<CLuaVector4D>();
+        return (NextCouldBeNumber(iOffset * 3) && NextCouldBeNumber(iOffset * 3 + 1) && NextCouldBeNumber(iOffset * 3 + 2)) || NextIsUserDataOfType<CLuaVector3D>(iOffset * 3) ||
+               NextIsUserDataOfType<CLuaVector4D>(iOffset * 3);

```

Comment:
```suggestion
        iOffset *= 3;
        return (NextCouldBeNumber(iOffset) && NextCouldBeNumber(iOffset + 1) && NextCouldBeNumber(iOffset + 2)) || NextIsUserDataOfType<CLuaVector3D>(iOffset) ||
               NextIsUserDataOfType<CLuaVector4D>(iOffset);
```

this won't make it more efficient, but it makes the code cleaner.